ui/appui-react/src/appui-react/statusfields/SnapMode.tsx

```diff
@@ -91,77 +106,46 @@ interface SnapModeFieldProps extends CommonProps {
  */
 export function SnapModeField(props: SnapModeFieldProps) {
   const { translate } = useTranslation();
+  // The snap modes displayed in the menu. Filtering out the snap modes that are not available.
+  const snapModeEntries: SnapModeFieldEntry[] = allSnapModeFieldEntries.filter(
+    (entry) => {
+      return props.availableSnapModes
+        ? props.availableSnapModes.some(
+            (availableSnap) => availableSnap === (entry.value as SnapMode)
+          )
+        : true;
+    }
+  );
+
```

Comment:
If `props.availableSnapModes` is not provided then there is no reason to filter the entries. You can move that if condition outside of the filter.